### PR TITLE
Rule: Element with `lang` attribute has valid language tag - TF feedback amendments

### PR DESCRIPTION
**Pull request by jeeyyy**
Tackled TF feedback provided in [issue 431](https://github.com/w3c/wcag-act/issues/431)

Closes issue(s):

- https://github.com/w3c/wcag-act/issues/431

Need for Final Call:
- 1 week

### CODE REVIEW

File: _rules/element-lang-valid-de46e4.md [hunk 1]
```diff
@@ -27,20 +27,21 @@ acknowledgments:
 
 This rules applies to any HTML element that:
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:28]: This needs to follow the new structure:

```suggestion
This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all the following is true:
```

File: _rules/element-lang-valid-de46e4.md [hunk 2]
```diff
@@ -27,20 +27,21 @@ acknowledgments:
...
 
-- has a [node document](https://dom.spec.whatwg.org/#concept-node-document) with a [content type](https://dom.spec.whatwg.org/#concept-document-content-type) of `text/html`; and
-- is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of a `body` element; and
-- has a `lang` attribute that is not empty ("").
+- has a [node document][] with a [content type][] of `text/html`; and
+- is a [descendant][] in the [flat tree][] of a `body` element; and
+- has a `lang` attribute that is not empty (`""`).
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:32]:
```suggestion
- the element is a [descendant][] in the [flat tree][] of a `body` element; and
- the element has a [node document][] with a [content type][] of `text/html`.
```

File: _rules/element-lang-valid-de46e4.md [hunk 3]
```diff
@@ -27,20 +27,21 @@ acknowledgments:
...
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:33]: Please note that the TF suggested that elements without text content not be applicable. Definitely belongs in a different PR, but that does mean this PR doesn't fully address the feedback from the TF.

File: _rules/element-lang-valid-de46e4.md [hunk 4]
```diff
@@ -27,20 +27,21 @@ acknowledgments:
...
 
 ## Expectation
 
 For each test target, the value of the `lang` attribute is a [valid language tag][].
 
 ## Assumptions
 
-- The `lang` attribute is assumed to be used to indicate the language of a section of the content. If the `lang` attribute is used for something else (for example to indicate a `code` element contains CSS), the content may still conform to WCAG despite failing this rule.
+- The language of the page can be set by other methods than the `lang` attribute, for example using HTTP headers or the `meta` element. These methods are not supported by all assistive technologies. This rule assumes that these other methods are insufficient to satisfying [Success Criterion 3.1.1: Language of Page][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:40]: This rule is about sections of content. I don't see how this is relevant.

File: _rules/element-lang-valid-de46e4.md [hunk 5]
```diff
@@ -59,24 +60,28 @@ There are differences in how assistive technologies handle unknown and invalid l
...
 
-The `lang` attribute has a value that is not empty ("") and has a valid language tag. The region section in the value is ignored by the rule.
+The `lang` attribute has a value that is not empty (`""`) and has a valid language tag. The region section in the value is ignored by the rule.
 
 ```html
 <html>
 	<body>
-		<blockquote lang="fr-CH"></blockquote>
+		<blockquote lang="fr-CH">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:83]: This needs to be French, otherwise it doesn't pass the SC.

File: _rules/element-lang-valid-de46e4.md [hunk 6]
```diff
@@ -102,32 +109,36 @@ The `lang` attribute value is not a primary language subtag.
 ```html
 <html>
 	<body>
-		<article lang="dutch"></article>
+		<article lang="dutch">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:113]:
```suggestion
			Zij liepen een vreemde Tiki bar binnen, aan de rand van een dorpje aan het strand.
```

File: _rules/element-lang-valid-de46e4.md [hunk 7]
```diff
@@ -25,22 +25,22 @@ acknowledgments:
 
 ## Applicability
 
-This rules applies to any HTML element that:
+This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all the following is true:
```
Comment (EmmaJP) [inline _rules/element-lang-valid-de46e4.md:28]:
```suggestion
This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all the following are true:
```

grammar

File: _rules/element-lang-valid-de46e4.md (hunk 3 above)
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:32]: Could that not be an additional condition within this list, added with this PR?

File: _rules/element-lang-valid-de46e4.md (hunk 5 above)
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:82]:
```suggestion
			Les livres sont des amis froids et sûrs.
```

A little Victor Hugo perhaps?

File: _rules/element-lang-valid-de46e4.md (hunk 4 above)
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:39]: Agree with Wilco that this assumption is less relevant for this rule. Even if the page language is not set, a `lang` attribute on a page element would have relevance and would need to be valid. That `lang` attribute should indicate the language of everything within the element the attribute is on.

File: _rules/element-lang-valid-de46e4.md (hunk 3 above)
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:33]: I added this as an additional condition within this list.

File: _rules/element-lang-valid-de46e4.md [hunk 8]
```diff
@@ -25,22 +25,21 @@ acknowledgments:
...
 ## Applicability
 
-This rules applies to any HTML element that:
+This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all of the following is true:
 
-- has a [node document](https://dom.spec.whatwg.org/#concept-node-document) with a [content type](https://dom.spec.whatwg.org/#concept-document-content-type) of `text/html`; and
-- is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of a `body` element; and
-- has a `lang` attribute that is not empty ("").
+- the element is a [descendant][] in the [flat tree][] of a `body` element; and
+- the element has a [node document][] with a [content type][] of `text/html`.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:31]:
```suggestion
- the element has a [node document][] with a [content type][] of `text/html`; and
```

File: _rules/element-lang-valid-de46e4.md [hunk 9]
```diff
@@ -25,22 +25,21 @@ acknowledgments:
...
+- the element has a [visible][] [text node][] as a [descendant][] in the [flat tree][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:32]: I think this should be in line with what I did here:

https://github.com/act-rules/act-rules.github.io/pull/1334/files#diff-53f178e37d6189f50875615ad27cecddR20

```suggestion
- the element has a [text node][] as a [descendant][] in the [flat tree][] that is [visible][] or [included in the accessibility tree][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 10]
```diff
@@ -25,22 +25,21 @@ acknowledgments:
...
 
 ## Expectation
 
 For each test target, the value of the `lang` attribute is a [valid language tag][].
 
 ## Assumptions
 
-- The `lang` attribute is assumed to be used to indicate the language of a section of the content. If the `lang` attribute is used for something else (for example to indicate a `code` element contains CSS), the content may still conform to WCAG despite failing this rule.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:40]: Why was this removed?

File: _rules/element-lang-valid-de46e4.md [hunk 11]
```diff
@@ -102,37 +107,57 @@ The `lang` attribute value is not a primary language subtag.
...
+		<article lang="  ">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
 		</article>
 	</body>
 </html>
 ```
 
+#### Failed Example 4
+
+The `lang` attribute value has a valid language tag, but does not have a [visible][] [text node][] as a [descendant][] in the [flat tree][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:147]: Wouldn't that make this inapplicable?

**Review by WilcoFiers — CHANGES_REQUESTED** [review]

File: _rules/element-lang-valid-de46e4.md (hunk 11 above)
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:147]: With the additional applicability - yes, and I think should continue to be included as an inapplicable example.

File: _rules/element-lang-valid-de46e4.md (hunk 10 above)
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:40]: +1

**Review by EmmaJP — CHANGES_REQUESTED** [review]
Agree with the changes Wilco has requested.

File: _rules/element-lang-valid-de46e4.md (hunk 10 above)
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:40]: One of the previous comments was citing that this is irrelevant - https://github.com/act-rules/act-rules.github.io/pull/1383#discussion_r452141399, hence removed it.

Let me know if I should add it back.

**Review by EmmaJP — APPROVED** [review]
The question around the removed assumption paragraph is outstanding, but I'm happy with however you and Wilco resolve that.

File: _rules/element-lang-valid-de46e4.md (hunk 10 above)
Comment (WilcoFiers) [reply _rules/element-lang-valid-de46e4.md:40]: I think this should be put back.

File: _rules/element-lang-valid-de46e4.md [hunk 12]
```diff
@@ -25,22 +25,21 @@ acknowledgments:
...
 
-This rules applies to any HTML element that:
+This rules applies to any HTML element with a `lang` [attribute value][] that is not empty (`""`) and for which all of the following is true:
 
-- has a [node document](https://dom.spec.whatwg.org/#concept-node-document) with a [content type](https://dom.spec.whatwg.org/#concept-document-content-type) of `text/html`; and
-- is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of a `body` element; and
-- has a `lang` attribute that is not empty ("").
+- the element is a [descendant][] in the [flat tree][] of a `body` element; and
+- the element has a [node document][] with a [content type][] of `text/html`; and
+- the element has a [text node][] as a [descendant][] in the [flat tree][] that is [visible][] or [included in the accessibility tree][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:32]: Recommend adding failed examples, one for visible but not in the accessibility tree, and one for not visible but still in the accessibility tree.

**Review by WilcoFiers — DISMISSED** [review]

**Review by WilcoFiers — DISMISSED** [review]

File: _rules/element-lang-valid-de46e4.md [hunk 13]
```diff
@@ -141,22 +184,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
+
+#### Inapplicable Example 3
+
+The `lang` attribute value has a valid language tag, but does not have a [visible][] [text node][] as a [descendant][] in the [flat tree][].
 
 ```html
 <html>
 	<body>
-		<article lang=""></article>
+		<p lang="en">
```
[the remark this answers is not in the document]
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:214]: @WilcoFiers I do not follow what you mean there. The example aims to show that a node with valid language tag  but no descendant text nodes is not applicable. 
What are you suggesting?

[the remark this answers is not in the document]
Comment (WilcoFiers) [reply _rules/element-lang-valid-de46e4.md:214]: What I'm suggesting is in my suggestion ;-). What this test case should demonstrate is that invalid lang attributes shouldn't fail if there's no text in them.

**Review by WilcoFiers — DISMISSED** [review]

File: _rules/element-lang-valid-de46e4.md [hunk 14]
```diff
@@ -97,37 +104,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+				They wandered into a strange Tiki bar on the edge of the small beach town.
+			</p>
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 5
+
+The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [visible][], it is still [included in the accessibility tree][].
```
Comment (EmmaJP) [inline _rules/element-lang-valid-de46e4.md:165]:
```suggestion
The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [visible][], though it is still [included in the accessibility tree][].
```

Minor grammar suggestion

File: _rules/element-lang-valid-de46e4.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (EmmaJP) [reply _rules/element-lang-valid-de46e4.md:214]: While the text display value is none, it doesn't matter what the lang attribute value is. It is inapplicable for testing, according to the applicability above. 

If the text display value were to change for some reason, then the validity of the lang value would become relevant. So while the lang value could be anything in this example, I think having an invalid value would make this example of something inapplicable for this rule less clear.

**Review by EmmaJP — DISMISSED** [review]
One minor grammar suggestion that doesn't affect meaning. Otherwise good.

**Review by WilcoFiers — APPROVED** [review]

File: _rules/element-lang-valid-de46e4.md [hunk 15]
```diff
@@ -59,36 +60,42 @@ There are differences in how assistive technologies handle unknown and invalid l
...
+			Ils se sont promenés dans un étrange bar Tiki en bordure de la petite ville balnéaire.
+		</blockquote>
 	</body>
 </html>
 ```
 
 #### Passed Example 3
 
-The `lang` attribute value has a valid language tag, but a syntactically invalid region subtag.
+The `lang` [attribute value][] has a valid language tag, but a syntactically invalid region subtag.
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:91]:
```suggestion
The `lang` [attribute value][] has a valid language tag, but a syntactically invalid region subtag which is ignored by the rule.
```

File: _rules/element-lang-valid-de46e4.md [hunk 16]
```diff
@@ -97,37 +104,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+		<article lang="  ">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 4
+
+The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [included in the accessibility tree][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:149]: So it is inapplicable, right?

File: _rules/element-lang-valid-de46e4.md [hunk 17]
```diff
@@ -97,37 +104,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+				They wandered into a strange Tiki bar on the edge of the small beach town.
+			</p>
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 5
+
+The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [visible][] though it is still [included in the accessibility tree][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:165]: This is also inapplicable, right?

File: _rules/element-lang-valid-de46e4.md [hunk 18]
```diff
@@ -141,22 +184,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
+		<article lang="">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 3
+
+The `lang` [attribute value][] has a valid language tag, but does not have a [visible][] [text node][] as a [descendant][] in the [flat tree][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:209]: Is "hidden" a valid language tag?

**Review by carlosapaduarte — DISMISSED** [review]
You need to add CSS Styling and Accessibility Tree to the input aspects (for the visibility and inclusion in the accessibility tree checks)

File: _rules/element-lang-valid-de46e4.md (hunk 18 above)
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:209]: @carlosapaduarte this [comment](https://github.com/act-rules/act-rules.github.io/pull/1383#discussion_r494234889) and the one prior to it should help.

File: _rules/element-lang-valid-de46e4.md (hunk 16 above)
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:149]: I believe the applicability does a play in words, where it is applicable for either visible or included in accessibility tree. So that is what the below 2 examples are trying to show.

File: _rules/element-lang-valid-de46e4.md (hunk 17 above)
Comment (jeeyyy) [reply _rules/element-lang-valid-de46e4.md:165]: See - https://github.com/act-rules/act-rules.github.io/pull/1383#discussion_r501678774

File: _rules/element-lang-valid-de46e4.md (hunk 16 above)
Comment (carlosapaduarte) [reply _rules/element-lang-valid-de46e4.md]: You are absolutely correct sir!

File: _rules/element-lang-valid-de46e4.md [hunk 19]
```diff
@@ -97,37 +106,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+		<article lang="  ">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 4
+
+The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [included in the accessibility tree][] though it is still [visible][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:151]: You need to update the description. If the language tag was valid then this would be a pass ;-)

File: _rules/element-lang-valid-de46e4.md [hunk 20]
```diff
@@ -97,37 +106,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+				They wandered into a strange Tiki bar on the edge of the small beach town.
+			</p>
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 5
+
+The `lang` [attribute value][] has a valid language tag, but its [descendant][] is not [visible][] though it is still [included in the accessibility tree][].
```
Comment (carlosapaduarte) [inline _rules/element-lang-valid-de46e4.md:167]: Same here. If the language tag is valid, this is a pass!

File: _rules/element-lang-valid-de46e4.md (hunk 18 above)
Comment (carlosapaduarte) [reply _rules/element-lang-valid-de46e4.md:211]: I understand @WilcoFiers point, and I would perhaps side with @EmmaJP that having an invalid value makes the example less clear. 
Nevertheless, the example description cannot state the language tag is valid because it is not :-)

**Review by carlosapaduarte — CHANGES_REQUESTED** [review]

**Review by carlosapaduarte — DISMISSED** [review]

**Comment by jeeyyy** [comment]
Final call ends October 23rd.

File: _rules/element-lang-valid-de46e4.md [hunk 21]
```diff
@@ -97,37 +106,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+		<article lang="  ">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 4
+
+The `lang` [attribute value][] does not have a valid language tag, and its [descendant][] is not [included in the accessibility tree][] though it is still [visible][].
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:151]: Being included in the accessibility tree has nothing to do with the Expectation.
This description makes it look like this is a reason for failure. This is not the case.

If the idea of this (and following example) is to show that Applicability is a "or" (visible or accessible), I'd move them to Passed Example (updating code) with a description pointing that ("it is not included in the accessibility tree but is still visible, hence it is Applicable").

File: _rules/element-lang-valid-de46e4.md [hunk 22]
```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:186]: Update description in the line of "there is no element with a `lang` attribute value which is a descendant of a `body` element".

File: _rules/element-lang-valid-de46e4.md [hunk 23]
```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
+	<body>
+		They wandered into a strange Tiki bar on the edge of the small beach town.
+	</body>
 </html>
 ```
 
 #### Inapplicable Example 2
 
-An empty value for the `lang` attribute is ignored, as the rule only applies to `lang` attributes that are not empty ("").
+An empty `lang` [attribute value][] is ignored, as the rule only applies to `lang` attributes that are not empty (`""`).
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:197]:
```suggestion
This `article` element has an empty `lang` [attribute value][] and is thus ignored by the rule.
```
or
```suggestion
There is no element with a non-empty `lang` [attribute value][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 24]
```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
+		<article lang="">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 3
+
+The `lang` [attribute value][] does not have a valid language tag, and does not have a [visible][] [text node][] as a [descendant][] in the [flat tree][].
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:211]:
```suggestion
This `p` element does not have a [text node][] as a [descendant][] in the [flat tree][] that is either [visible][] or [included in the accessibility tree][].
```
or 
```suggestion
There is no element with a [text node][] as a [descendant][] in the [flat tree][] that is either [visible][] or [included in the accessibility tree][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 25]
```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
 
 ```html
 <html>
 	<body>
-		<article lang=""></article>
+		<p lang="hidden">
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:216]: Use a valid language. Having an invalid one does not bring anything to that example (it's about visible/accessible text).
```suggestion
		<p lang="en">
```

File: _rules/element-lang-valid-de46e4.md [hunk 26]
```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
...
+[flat tree]: https://drafts.csswg.org/css-scoping/#flat-tree
 [grandfathered tags]: https://tools.ietf.org/html/bcp47#section-2.2.8
 [bcp 47]: https://tools.ietf.org/html/bcp47#section-2.1
 [valid language tag]: #valid-language-tag
+[whitespace]: #whitespace 'Definition of Whitespace'
+[visible]: #visible 'Definition of visible'
+[text node]: https://dom.spec.whatwg.org/#text
+[descendant]: https://dom.spec.whatwg.org/#concept-tree-descendant
+[attribute value]: #attribute-value 'Definition of Attribute Value'
+[included in the accessibility tree]: #included-in-the-accessibility-tree
```
Comment (Jym77) [inline _rules/element-lang-valid-de46e4.md:237]:
```suggestion
[attribute value]: #attribute-value 'Definition of Attribute Value'
[bcp 47]: https://tools.ietf.org/html/bcp47#section-2.1
[content type]: https://dom.spec.whatwg.org/#concept-document-content-type
[descendant]: https://dom.spec.whatwg.org/#concept-tree-descendant
[flat tree]: https://drafts.csswg.org/css-scoping/#flat-tree
[grandfathered tags]: https://tools.ietf.org/html/bcp47#section-2.2.8
[included in the accessibility tree]: #included-in-the-accessibility-tree
[node document]: https://dom.spec.whatwg.org/#concept-node-document
[text node]: https://dom.spec.whatwg.org/#text
[valid language tag]: #valid-language-tag
[visible]: #visible 'Definition of visible'
[whitespace]: #whitespace 'Definition of Whitespace'
```
If it had been ordered in the first place, you'd probably have noticed the duplicate reference to "descendant" and clean it…

**Review by Jym77 — DISMISSED** [review]
Editorial and polishing.

File: _rules/element-lang-valid-de46e4.md [hunk 27]
```diff
@@ -25,22 +27,23 @@ acknowledgments:
...
-- is a [descendant](https://dom.spec.whatwg.org/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of a `body` element; and
-- has a `lang` attribute that is not empty ("").
+- the element is a [descendant][] in the [flat tree][] of a `body` element; and
+- the element has a [node document][] with a [content type][] of `text/html`; and
+- the element has a [text node][] as a [descendant][] in the [flat tree][] that is [visible][] or [included in the accessibility tree][].
 
 ## Expectation
 
-For each test target, the value of the `lang` attribute is a [valid language tag][].
+For each test target, the value of the `lang` [attribute value][] is a [valid language tag][].
```
[the remark this answers is not in the document]
Comment (WilcoFiers) [reply _rules/element-lang-valid-de46e4.md:38]: Oof! Good catch!

File: _rules/element-lang-valid-de46e4.md (hunk 25 above)
Comment (WilcoFiers) [reply _rules/element-lang-valid-de46e4.md:216]: Right but then this does not prove the case this test is for; that an invalid lang attribute should not fail if it's on a hidden element.

File: _rules/element-lang-valid-de46e4.md [hunk 28]
```diff
@@ -97,37 +106,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+		<article lang="  ">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Failed Example 4
+
+The `lang` [attribute value][] does not have a valid language tag, and its [descendant][] is not [included in the accessibility tree][] though it is still [visible][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:151]:
```suggestion
The `lang` [attribute value][] does not have a valid language tag. The `lang` attribute must be valid because the content is [visible][].
```

File: _rules/element-lang-valid-de46e4.md [hunk 29]
```diff
@@ -59,36 +62,42 @@ There are differences in how assistive technologies handle unknown and invalid l
...
+			Ils ont trouvé un étrange bar Tiki aux abords de la petite ville balnéaire.
+		</blockquote>
 	</body>
 </html>
 ```
 
 #### Passed Example 3
 
-The `lang` attribute value has a valid language tag, but a syntactically invalid region subtag.
+This `p` element has a`lang` [attribute value][] which has a [valid language tag][], but a syntactically invalid region subtag which is ignored by the rule.
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:93]:
```suggestion
This `p` element has a `lang` [attribute value][] which has a [valid language tag][], but a syntactically invalid region subtag which is ignored by the rule.
```

**Review by Jym77 — APPROVED** [review]

**Review by carlosapaduarte — APPROVED** [review]

File: _rules/element-lang-valid-de46e4.md [hunk 30]
```diff
@@ -97,37 +106,73 @@ The `lang` attribute value has a valid language tag, but a syntactically invalid
...
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
 	</body>
 </html>
 ```
 
 #### Failed Example 3
 
-The `lang` attribute value consists of only [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace) and is not a [valid language tag][].
+This `article` element` has a `lang` [attribute value][] which consists of only [whitespace][] and thus is not a [valid language tag][].
```
Comment (WilcoFiers) [inline _rules/element-lang-valid-de46e4.md:137]:
```suggestion
This `article` element has a `lang` [attribute value][] which consists of only [whitespace][] and thus is not a [valid language tag][].
```

**Review by ShadowBB — APPROVED** [review]

**Comment by WilcoFiers** [comment]
Merging, since the changes made since the final call were editorial.
